kernel/kernel-api/src/main/java/io/delta/kernel/internal/commit/DefaultFileSystemManagedTableOnlyCommitter.java

```diff
@@ -39,7 +52,37 @@ public CommitResponse commit(
       throws CommitFailedException {
     commitMetadata.getReadProtocolOpt().ifPresent(this::validateProtocol);
     commitMetadata.getNewProtocolOpt().ifPresent(this::validateProtocol);
-    throw new UnsupportedOperationException("Default Committer not yet implemented");
+
+    final String jsonCommitFile =
+        FileNames.deltaFile(commitMetadata.getDeltaLogDirPath(), commitMetadata.getVersion());
+
+    logger.info("Attempting to commit {}", jsonCommitFile);
+
+    try {
+      wrapEngineExceptionThrowsIO(
+          () -> {
+            engine
+                .getJsonHandler()
+                .writeJsonFileAtomically(jsonCommitFile, finalizedActions, false /* overwrite */);
+            return null;
+          },
+          String.format("Write file actions to JSON log file `%s`", jsonCommitFile));
+    } catch (FileAlreadyExistsException e) {
+      throw new CommitFailedException(
+          true /* retryable */,
+          true /* conflict */,
+          "Concurrent write detected for version " + commitMetadata.getVersion(),
+          e);
+    } catch (IOException e) {
+      throw new CommitFailedException(
+          true /* retryable */,
+          false /* conflict */,
+          "Failed to write commit file due to I/O error: " + e.getMessage(),
+          e);
+    }
```

Comment:
And I guess we decided to just propagate other RuntimeExceptions?

Reply:
Yup! That's what the UC-Commit-Coordinator-Client does today! (for example)